### PR TITLE
fix: OPTIC-945: Improve the bundling strategy and ship a single entrypoint with a common vendor chunk across all libs - restore

### PR DESCRIPTION
**Pull request by yyassi-heartex**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

### Describe the reason for change
we need to be able to work with one version of React

### CODE REVIEW

**Comment by yyassi-heartex** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10386085512)
> Successfully merged:  22 files changed, 364 insertions(+), 53 deletions(-)

File: web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx [hunk 1]
```diff
@@ -11,13 +11,16 @@ import { Phrases } from "./Phrases";
...
+import { cn } from "../../../utils/bem";
 
 const audioDefaultProps = {};
 
 if (isFF(FF_LSDV_4711)) audioDefaultProps.crossOrigin = "anonymous";
 
 class HtxParagraphsView extends Component {
   _regionSpanSelector = ".htx-highlight";
+  mainContent = `.${cn("main-content").toClassName()}`;
+  mainViewAnnotation = `.${cn("main-view").elem("annotation").toClassName()}`;
```
Comment (bmartel) [inline web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx]:
```suggestion
  mainContentSelector = `.${cn("main-content").toClassName()}`;
  mainViewAnnotationSelector = `.${cn("main-view").elem("annotation").toClassName()}`;
```

File: web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx [hunk 2]
```diff
@@ -510,12 +513,12 @@ class HtxParagraphsView extends Component {
...
   componentDidMount() {
     if (isFF(FF_LSDV_E_278) && this.props.item.contextscroll)
-      this._resizeObserver.observe(document.querySelector(".lsf-main-content"));
+      this._resizeObserver.observe(document.querySelector(this.mainContent));
     this._handleUpdate();
   }
 
   componentWillUnmount() {
-    const target = document.querySelector(".lsf-main-content");
+    const target = document.querySelector(this.mainContent);
```
Comment (bmartel) [inline web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx]:
```suggestion
    const target = document.querySelector(this.mainContentSelector);
```

File: web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx [hunk 3]
```diff
@@ -510,12 +513,12 @@ class HtxParagraphsView extends Component {
 
   componentDidMount() {
     if (isFF(FF_LSDV_E_278) && this.props.item.contextscroll)
-      this._resizeObserver.observe(document.querySelector(".lsf-main-content"));
+      this._resizeObserver.observe(document.querySelector(this.mainContent));
```
Comment (bmartel) [inline web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx]:
```suggestion
      this._resizeObserver.observe(document.querySelector(this.mainContentSelector));
```

File: web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx [hunk 4]
```diff
@@ -484,10 +487,10 @@ class HtxParagraphsView extends Component {
   _handleScrollContainerHeight = () => {
     requestAnimationFrame(() => {
       const container = this.myRef.current;
-      const mainContentView = document.querySelector(".lsf-main-content");
+      const mainContentView = document.querySelector(this.mainContent);
       const mainRect = mainContentView.getBoundingClientRect();
       const visibleHeight = document.documentElement.clientHeight - mainRect.top;
-      const annotationView = document.querySelector(".lsf-main-view__annotation");
+      const annotationView = document.querySelector(this.mainViewAnnotation);
```
Comment (bmartel) [inline web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx]:
```suggestion
      const annotationView = document.querySelector(this.mainViewAnnotationSelector);
```

File: web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx [hunk 5]
```diff
@@ -484,10 +487,10 @@ class HtxParagraphsView extends Component {
   _handleScrollContainerHeight = () => {
     requestAnimationFrame(() => {
       const container = this.myRef.current;
-      const mainContentView = document.querySelector(".lsf-main-content");
+      const mainContentView = document.querySelector(this.mainContent);
```
Comment (bmartel) [inline web/libs/editor/src/tags/object/Paragraphs/HtxParagraphs.jsx]:
```suggestion
      const mainContentView = document.querySelector(this.mainContentSelector);
```

**Review by bmartel — APPROVED** [review]
Nicely done! Though we can't yet achieve the better prefix scenario of `ls-` as had been hoped without breaking backwards compatibility, this is a very low touch alternative which allows us to move forward! With the other initiatives and refactors we can iterate this as well to get towards the ideal.

**Comment by vladimirheartex** [comment]
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10472879183)
> Docker image was pushed with the tag `20240820.135058-fb-optic-945-restore-f6aea07ed`

**Comment by vladimirheartex** [comment]
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10485295977)
> Docker image was pushed with the tag `20240821.071738-fb-optic-945-restore-e8c2d9c29`

**Comment by vladimirheartex** [comment]
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10555837384)
> Docker image was pushed with the tag `20240826.075349-fb-optic-945-restore-255af8b2a`

**Comment by vladimirheartex** [comment]
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10574820416)
> Docker image was pushed with the tag `20240827.085026-fb-optic-945-restore-64df9f7db`

**Comment by vladimirheartex** [comment]
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10592770927)
> **Error**: failed to execute "build" command

**Comment by vladimirheartex** [comment]
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10593169156)
> **Error**: failed to execute "build" command

**Comment by vladimirheartex** [comment]
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/10595888588)
> Docker image was pushed with the tag `20240828.113838-fb-optic-945-restore-9998365a1`

**Review by MihajloHoma — APPROVED** [review]

**Pull request merged**